File: Binary Tree inorder traversal/Go/solution1.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func Main() {}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
	result := make([]int, 0)
	if root == nil {
		return result
	}

	if root.Left != nil {
		arr := inorderTraversal(root.Left)
		result = append(result, arr...)
	}

	result = append(result, root.Val)

	if root.Right != nil {
		arr := inorderTraversal(root.Right)
		result = append(result, arr...)
	}

	return result
}

File: Binary Tree inorder traversal/C#/solution1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        List<int> vals = new List<int>();
        traversal(root, vals);
        return vals;
    }
    
    public void traversal(TreeNode root, List<int> vals){
        if(root == null){
            return;
        } else {
            traversal(root.left, vals);
            vals.Add(root.val);
            traversal(root.right, vals);
        }
    }
}